Sanity check using new compilers

mhrib opened the pull request:
Code is checked using new intel/gnu/cray compilers on cray XC50 HPC.

Corrections:
- Do loops has to end with an END DO or CONTINUE. Some compilers now treats this old f77 syntax as errors.
- Variables with same name changes between scalar and vectors in between subroutines
- Out-of-bounds
- OMP directives
- 'etime' routine replaced by fortran standard 'cpu_time'. However not used.
- Other minor stuff

Added:
- config/DMI_xc50_{intel,gnu,cray}_setup : Compiler flags for Cray XC50 hpc

New feature:
- archive/src/getdat.f : Can handle arch{v,s} files with/without 'covice' and 'thkice'

awallcraft commented:
The _setup config files (and bin/Make_all.csh) need to include a compiler option for reading and writing bigendian files.

Your DMI_xc50_intel_setup is identical to xc30IF_setup, except that it is missing "-convert big_endian".  I suggest renaming it xc50-intel_setup and adding the missing compiler option.

Your DMI_xc50_gnu_setup is similar to intelGF_setup except that it is missing "-fconvert=big-endian". I suggest renaming it xc50-gnu_setup and adding the missing compiler option.  Also, ifort on the 2nd line should be gfortran.

I don't know what the CCE compiler option for bigendian I/O is, but if you want to include this setup file rename it xc50-cce_setup and add the appropriate compiler option. Otherwise delete this file.

Make the same changes to bin/Make_all.csh and bin/Make_ncdf.csh.

Finally, ALT/bin is a copy of bin that is present in case you need to use a different compiler.  Make all your changes that are in bin to ALT/bin.

mhrib commented:
1) Changed ALT/bin reflecting the changes in bin directory.

2) To my knowledge, there is no specific CCE compiler option for big-endian I/O. However "-h byteswapio" does usually swap to big-endian, as systems is usually little-endian.

3) It was by purpose, that the config/xc50-\<compiler\>\_setup files use native endian (usually little endian), as this is what we're doing with HYCOM-src and HYCOM-tools. And similar with bin/Make_{all,ncdf}.csh.  I have added Information about that choice in top of the config/xc50_\<compiler\>\_setup files. And also leaving the "big-endian" flags right above.
**Can this be a solution**?
If not, I will change them to the "big endian" versions and we will handle the few changes internally at our institute. Not a big deal for us.

4) config/xc50-**cce**\_setup renamed to config/xc50-**cray**\_setup
This notation follows the "module" environment:
`module load PrgEnv-cray`
`env | grep ^PE_ENV | cut -f2 -d= | tr "[:upper:]" "[:lower:]"` ( = cray )
